data/data-pipeline/data_pipeline/etl/score/etl_score_post.py

```diff
@@ -159,19 +163,25 @@ def _create_score_data(
         states_df: pd.DataFrame,
         score_df: pd.DataFrame,
     ) -> pd.DataFrame:
-        # merge state with counties
-        logger.info("Merging state with county info")
-        county_state_merged = counties_df.merge(
-            states_df, on="State Abbreviation", how="left"
-        )
 
-        # merge state + county with score
-        score_county_state_merged = score_df.merge(
-            county_state_merged,
+        logger.info("Merging county info with score info")
+        score_county_merged = score_df.merge(
+            # We drop state abbreviation so we don't get it twice
```

Comment:
The crucial difference between this and the original code is the order of the joins. In the original code, we merged state and county, and then merged from state and county to score based on GEOID. Island territories, however, do not have GEOIDs. To allow them to retain the state information, we merge counties first and then merge states. This is slightly annoying from the perspective of an n_tracts join two times.

Reply:
I'm sorry, I'm not following this. "Island territories, however, do not have GEOIDs" - ? Do you mean they don't have counties?

Reply:
yes! that's exactly what I mean. Because they don't have counties, in the `state_county_merged_df`, there are nulls for `GEOID`. Thus, when we join on `GEOID`, we drop the island area information. Does this make more sense? I'll update the comment inline.